### PR TITLE
Minor: remove null in `array-append` and `array-prepend`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Related #8625 #8317 Thanks @jayzhan211 effort, we can remove `NULL` in `array-append` and `array-prepend`

### CODE REVIEW

File: datafusion/sqllogictest/test_files/array.slt [hunk 1]
```diff
@@ -1577,9 +1577,9 @@ select
 query ????
 select
   array_append(arrow_cast(make_array(), 'LargeList(Null)'), 4),
-  array_append(make_array(), null),
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/array.slt:1580]: tests without arrow-case looks valid for me, should we delete them?

Comment (jayzhan211) [reply datafusion/sqllogictest/test_files/array.slt:1580]: what is arrow-case?

**Review by jayzhan211 — APPROVED** [review]
LGTM

File: datafusion/sqllogictest/test_files/array.slt (hunk 1 above)
Comment (Weijun-H) [reply datafusion/sqllogictest/test_files/array.slt:1580]: Maybe arrow-cast?

Comment (comphead) [reply datafusion/sqllogictest/test_files/array.slt:1580]: right, arrow-cast, no worries, I found we already have such cases in `array.slt`

**Review by comphead — APPROVED** [review]
lgtm thanks again @Weijun-H

File: datafusion/physical-expr/src/array_expressions.rs [hunk 2]
```diff
@@ -1031,12 +1031,6 @@ where
     let res = match list_array.value_type() {
         DataType::List(_) => concat_internal::<i32>(args)?,
         DataType::LargeList(_) => concat_internal::<i64>(args)?,
-        DataType::Null => {
```
Comment (alamb) [inline datafusion/physical-expr/src/array_expressions.rs:1034]: 👨‍🍳 👌

**Pull request merged**